Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/RefreshableNode.java

```diff
@@ -133,38 +131,43 @@ public void run() {
                             Futures.addCallback(future, new FutureCallback<List<Node>>() {
                                 @Override
                                 public void onSuccess(List<Node> nodes) {
-                                    updateName(nodeName, null);
-                                    updateNodeNameAfterLoading();
-                                    expandNodeAfterLoading();
+                                    DefaultLoader.getIdeHelper().invokeAndWait(new Runnable() {
```

Comment:
Will `invokeAndWait` be blocked?

Reply:
Replaced with `invokeLater()` in the latest commit

---
In reply to: [242909424](https://github.com/Microsoft/azure-tools-for-java/pull/2534#discussion_r242909424) [](ancestors = 242909424)